src/pages/Environments.js

```diff
@@ -110,6 +111,11 @@ const DeleteAppModal = ({ app: { googleProject, appName, diskName, appType }, on
   ])
 }
 
+const addNamespaceLabelIfAbsent = resource => {
+  // Use googleProject if workspace namespace label is not defined
+  return _.merge({ labels: { saturnWorkspaceNamespace: resource.googleProject } }, resource)
```

Comment:
if a resource has `labels.saturnWorkspaceNamespace` then this will not do anything correct? would it be possible to only call this method in the case that `saturnWorkspaceNamespace` is undefined?

Reply:
Yea, totally. Thanks! Updated the code.